.buildkite/pipeline.nix

```diff
@@ -0,0 +1,172 @@
+    }
+    { label = "test nix-built binaries";
+      command = "nix-build tests/tezos-nix-binaries.nix --no-out-link";
+      only_changes = nix_changes_regexes;
+    }
+    { label = "build via docker";
+      commands = [
+        "eval \"$SET_VERSION\""
+        "cd docker"
+        "./docker-static-build.sh"
+        "nix run -f.. pkgs.upx -c upx tezos-*"
+      ];
+      artifact_paths = [
+        "./docker/tezos-*"
+      ];
+      agents = { queue = "docker"; };
+      only_changes = static_binaries_changes_regexes;
+    }
+    { label = "build arm via docker";
+      commands = [
+        "eval \"$SET_VERSION\""
+        "cd docker"
+        "./docker-static-build.sh"
+        "upx tezos-*"
+        "for f in ./tezos-*; do mv \"\$f\" \"\$f-arm64\"; done"
+      ];
+      artifact_paths = [
+        "./docker/tezos-*"
+      ];
+      agents = { queue = "arm64-build"; };
+      only_changes = static_binaries_changes_regexes;
+    }
+    { label = "check brew formulas";
+      commands = [
+        # Check all formulas except 'tezos.rb' because it's a base class for all other formulas
+        "find ./Formula -type f -name \"*.rb\" -exec ruby -c {} +"
+        # All formulas share the same source URL inherited by the Tezos class, so it's fine
+        # to fetch sources for only one formula
+        "brew fetch -s ./Formula/tezos-client.rb"
+      ];
+      agents = { queue = "x86_64-darwin"; };
+      only_changes = brew_changes_regexes;
+    }
+    "wait"
+    { label = "test docker-built binaries";
+      commands = [
+        "buildkite-agent artifact download \"docker/*\" . --step \"build via docker\""
+        "chmod +x ./docker/*"
+        "nix-build tests/tezos-binaries.nix --no-out-link --arg path-to-binaries ./docker"
+      ];
+      branches = "!master";
+      only_changes = static_binaries_changes_regexes;
+    }
+    { label = "test deb source packages via docker";
+      commands = [
+        "eval \"$SET_VERSION\""
+        "./docker/docker-tezos-packages.sh --os ubuntu --type source"
+      ];
+      artifact_paths = [
+        "./out/*"
+      ];
+      branches = "!master";
+      timeout_in_minutes = 60;
+      agents = { queue = "docker"; };
+      only_changes = native_packaging_changes_regexes;
+    }
+    { label = "test deb binary packages via docker";
+      commands = [
+        "eval \"$SET_VERSION\""
+        # Building all binary packages will take significant amount of time, so we build only one
+        # in order to ensure package generation sanity
+        "./docker/docker-tezos-packages.sh --os ubuntu --type binary --package tezos-baker-008-PtEdo2Zk"
+        "rm -rf out"
+      ];
+      artifact_paths = [
+        "./out/*"
+      ];
+      # It takes much time to build binary package, so we do it only on master
+      branches = "master";
+      timeout_in_minutes = 90;
+      agents = { queue = "docker"; };
+      only_changes = native_packaging_changes_regexes;
```

Comment:
There will actually be issues with steps that run on master branch, the diff will always be empty :pk:

Reply:
hmm, indeed, using `git diff` won't work this easy :(

Reply:
looks like `git diff --name-only HEAD HEAD~1` should be used in case `git diff --name-only origin/master...HEAD` is empty